src/main.rs

```diff
@@ -72,39 +75,138 @@ impl event::EventHandler for MainState {
             lane.update_vehicles_in_lane();
         }
 
+        //Check for game over
+        if self.player.get_lives() <= 0 {
+            self.player.set_lives();
+
+            //Clear screen, optional
+            graphics::clear(_ctx);
+
+            //Gamve over has a scalable center, text should always be in center regardless of dimensions
+            let center:f32 = WIN_W as f32 / 2.0 - *&self.game_over_man.width() as f32 / 2.0;
+
+            let dest_point = graphics::Point2::new(center, WIN_H as f32 / 2.0);
+            graphics::draw(_ctx, &self.game_over_man, dest_point, 0.0)?;
+            graphics::present(_ctx);
+            timer::sleep(Duration::from_secs(2));
+        }
+
+        //Take a life
+        if self.player.get_life_lost() == true {
+            self.player.set_life_lost();
+            timer::sleep(Duration::from_secs(1));
+            self.player.restart_x();
+            self.player.restart_y();
+        }
+
         Ok(())
     }
 
     fn draw(&mut self, ctx: &mut Context) -> GameResult<()> {
         graphics::clear(ctx);
 
-        //Draw background
-        self.road.draw(ctx)?;
-        self.river.draw(ctx)?;
-        // self.cubbie.draw(ctx)?;
-
-
-        //Draw our lanes
-        for lane in &mut self.lanes {
-            lane.draw_vehicles_in_lane(ctx)?;
+        if self.main_menu {
```

Comment:
Would it be painful/still work if the main menu work was moved into its own module? It almost seems like you've grouped the work into what could be functions in a module:`draw_crabber_name_upper_middle`, `draw_crab_logo`, etc.

Reply:
I moved the implementations into a new struct Menu in the Background module